packages/server/src/index.ts

```diff
@@ -2,47 +2,57 @@ import { createServer } from 'node:http';
 import { createYoga } from 'graphql-yoga';
 import 'reflect-metadata';
 import { useGraphQLModules } from '@envelop/graphql-modules';
-import { useHive } from '@graphql-hive/yoga';
 import { useDeferStream } from '@graphql-yoga/plugin-defer-stream';
-import { AccounterContext } from '@shared/types';
+import { AUTH_ROLE, AUTH_USER_ID, AUTH_USERNAME } from '@shared/constants';
+import { Role } from '@shared/gql-types';
+import { AccounterContext, UserType } from '@shared/types';
 import { env } from './environment.js';
 import { createGraphQLApp } from './modules-app.js';
 import { adminContextPlugin } from './plugins/admin-context-plugin.js';
-import { authPlugin } from './plugins/auth-plugin.js';
 
 async function main() {
   const application = await createGraphQLApp(env);
 
+  // While createGraphQLApp initializes the application, we use buildSubgraphSchema to build the Subgraph Schema.
   const yoga = createYoga({
-    plugins: [
-      authPlugin(),
-      adminContextPlugin(),
-      useGraphQLModules(application),
-      useDeferStream(),
-      useHive({
-        enabled: !!env.hive.hiveToken,
-        token: env.hive.hiveToken ?? '',
-        usage: true,
-      }),
-    ],
+    schema: application.schema,
+    graphqlEndpoint: '/subgraphs/legacy',
+    graphiql: {
+      title: 'Legacy Subgraph',
+    },
+    landingPage: true,
+    plugins: [adminContextPlugin(), useGraphQLModules(application), useDeferStream()],
     context: (yogaContext): AccounterContext => {
+      const headers = yogaContext.request.headers;
+      const authRole = headers.get(AUTH_ROLE) as Role;
+      const authUsername = headers.get(AUTH_USERNAME);
+      const authUserId = headers.get(AUTH_USER_ID);
+      if (!authRole || !authUsername || !authUserId) {
+        throw new Error('Unauthorized');
```

Comment:
Note to think about later: better error handling (see recently-introduced `LedgerLockError` on Main)